src/Log/LogService.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * This file is part of OPUS. The software OPUS has been originally developed
+ * at the University of Stuttgart with funding from the German Research Net,
+ * the Federal Department of Higher Education and Research and the Ministry
+ * of Science, Research and the Arts of the State of Baden-Wuerttemberg.
+ *
+ * OPUS 4 is a complete rewrite of the original OPUS software and was developed
+ * by the Stuttgart University Library, the Library Service Center
+ * Baden-Wuerttemberg, the Cooperative Library Network Berlin-Brandenburg,
+ * the Saarland University and State Library, the Saxon State Library -
+ * Dresden State and University Library, the Bielefeld University Library and
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Framework
+ * @package     Opus_Bootstrap
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2008-2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace Opus\Log;
+
+class logService
+{	
+
+	private $logFile;
+	public static $logger = array(); 
+    public $format;
+	private $config;
+	private $logFilePath = null;
+	private static $instance;
+	public $logFileName;
+
+	public static function getInstance(){ 
+
+		if(null === static::$instance){
+			static::$instance = new static();
+		}
+		return static::$instance;
+
+    }
+
+    public function openLog($configName, $logFileName){
+    	$this->getConfig($configName);	
+
+        $logfilePath = $this->logFilePath == null ? $this->setPath($logFileName) : $this->logFilePath;     
+
+        $this->logFileName = $logFileName;
+        $this->logFile = @fopen($logfilePath, 'a', false);  
+        if ($this->logFile === false) {
+            $path = dirname($logfilePath);
+
+            if (! is_dir($path)) {
+                throw new Exception('Directory for logging does not exist');
+            } else {
+                throw new Exception('Failed to open logging file:' . $logfilePath);
+            }
+        }
+
+    }
+
+    public function getConfig($configName){
```

Comment:
This is not going to work. There is only "Zend_Log" in the Zend_Registry, so getConfig() should be enough. However from that global configuration you need to get the config for a specific logger, so the function could handle that. It depends if you need the global config or just a subsection of it.

Reply:
Sorry, my mistake. It is still not going to work, but what you want is "Zend_Config". It is the global configuration for OPUS 4 read from `application.ini`, `config.ini` and `console.ini`. During tests there are also the `tests/config.ini` and the `tests/tests.ini`. The configuration for loggers would be a smaller part of the entire configuration.